crates/sozo/ops/src/account.rs

```diff
@@ -0,0 +1,417 @@
+                );
+            }
+
+            let mut factory = OpenZeppelinAccountFactory::new(
+                undeployed_status.class_hash,
+                chain_id,
+                signer,
+                &provider,
+            )
+            .await?;
+            factory.set_block_id(BlockId::Tag(BlockTag::Pending));
+
+            factory
+        }
+    };
+
+    let account_deployment = factory.deploy(undeployed_status.salt);
+
+    let target_deployment_address = account.deploy_account_address()?;
+
+    // Sanity check. We don't really need to check again here actually
+    if account_deployment.address() != target_deployment_address {
+        panic!("Unexpected account deployment address mismatch");
+    }
+
+    let max_fee = match fee_setting {
+        FeeSetting::Manual(fee) => MaxFeeType::Manual { max_fee: fee },
+        FeeSetting::EstimateOnly | FeeSetting::None => {
+            let estimated_fee = account_deployment
+                .estimate_fee()
+                .await
+                .map_err(|err| match err {
+                    AccountFactoryError::Provider(ProviderError::StarknetError(err)) => {
+                        map_starknet_error(err)
+                    }
+                    err => anyhow::anyhow!("{}", err),
+                })?
+                .overall_fee;
+
+            let estimated_fee_with_buffer = (estimated_fee * felt!("3")).floor_div(felt!("2"));
+
+            if fee_setting.is_estimate_only() {
+                println!("{} ETH", format!("{}", estimated_fee.to_big_decimal(18)).bright_yellow(),);
+                return Ok(());
+            }
+
+            MaxFeeType::Estimated {
+                estimate: estimated_fee,
+                estimate_with_buffer: estimated_fee_with_buffer,
+            }
+        }
+    };
+
+    if !simulate {
+        match max_fee {
+            MaxFeeType::Manual { max_fee } => {
+                eprintln!(
+                    "You've manually specified the account deployment fee to be {}. Therefore, \
+                     fund at least:\n    {}",
+                    format!("{} ETH", max_fee.to_big_decimal(18)).bright_yellow(),
+                    format!("{} ETH", max_fee.to_big_decimal(18)).bright_yellow(),
+                );
+            }
+            MaxFeeType::Estimated { estimate, estimate_with_buffer } => {
+                eprintln!(
+                    "The estimated account deployment fee is {}. However, to avoid failure, fund \
+                     at least:\n    {}",
+                    format!("{} ETH", estimate.to_big_decimal(18)).bright_yellow(),
+                    format!("{} ETH", estimate_with_buffer.to_big_decimal(18)).bright_yellow()
+                );
+            }
+        }
+
+        eprintln!(
+            "to the following address:\n    {}",
+            format!("{:#064x}", target_deployment_address).bright_yellow()
+        );
+
+        eprint!("Press [ENTER] once you've funded the address.");
```

Comment:
Add an option to skip this to allow testing and automated deployments.

Reply:
done